### PR TITLE
Add custom hub package & config hooks

### PR DESCRIPTION
**Pull request by yuvipanda**
- Add hook for installing additional hub packages
- Add hook for additional `jupyterhub_config.py`
  entries

 - [x] Add / update documentation
 - [x] Add tests

### CODE REVIEW

**Comment by yuvipanda** [comment]
Adding these hooks to help set up https://discourse.jupyter.org/t/would-a-the-littlest-binder-be-useful/1041/28.

File: integration-tests/test_simplest_plugin.py [hunk 1]
```diff
@@ -46,3 +53,11 @@ def test_config_hook():
         data = yaml.load(f)
 
     assert data['simplest_plugin']['present']
+
+def test_tmpauthenticator():
+    """
```
Comment (GeorgianaElena) [inline integration-tests/test_simplest_plugin.py]: I would probably name this *test_tmpauthenticator_enabled* or *test_jupyterhub_config_hook* (I'm not very good at naming things though :eyes: ) but the name as it is makes me think we're testing the authenticator.

File: tljh/installer.py [hunk 2]
```diff
@@ -208,6 +208,7 @@ def ensure_jupyterhub_package(prefix):
         'jupyterhub-firstuseauthenticator==0.12',
         'jupyterhub-nativeauthenticator==0.0.4',
         'jupyterhub-ldapauthenticator==1.2.2',
+        'jupyterhub-tmpauthenticator==0.6',
```
Comment (GeorgianaElena) [inline tljh/installer.py]: From what I understand, when a user wants to custom its juputerhub_config using *tljh_custom_jupyterhub_config* in order to use something that's not directly installed by TLJH, they could would use *tljh_extra_hub_pip_packages* to install it first. 

If this is the case, I believe tmpauthenticator package should/could come from *tljh_extra_hub_pip_packages*.

File: integration-tests/test_simplest_plugin.py (hunk 1 above)
Comment (yuvipanda) [reply integration-tests/test_simplest_plugin.py]: Good catch! I've renamed it!

File: tljh/installer.py (hunk 2 above)
Comment (yuvipanda) [reply tljh/installer.py]: I've split this part of the PR to https://github.com/jupyterhub/the-littlest-jupyterhub/pull/365 separately, and based this PR on that. So we have two PRs:

1. #365 that adds tmpauthenticator
2. This PR, which adds two hooks with tests for both of them. There is a test for `tljh_extra_hub_pip_packages` that imports the 'there' package, and the test for `tljh_custom_jupyterhub_config` that sets tmpauthenticator to be the default authenticator.

How does that sound?

**Comment by yuvipanda** [comment]
Thanks for the review, @GeorgianaElena! I've updated the PR to reflect your suggestions. Let me know what you think :)

#365 should probably be merged before this though - this PR 'depends' on that (it includes some commits from there). So once #365 gets merged, the diff on this PR should stop showing the contents of #365. This might be confusing - let me know if I can help clarify that.

**Comment by GeorgianaElena** [comment]
Excellent @yuvipanda :confetti_ball:. Once the tests pass I'll try to merge the PRs in the order you suggested :D

**Pull request merged**